Documentation/project-docs/linux-performance-tracing.md

```diff
@@ -0,0 +1,114 @@
+Performance Tracing on Linux
+============================
+
+When a performance problem is encountered on Linux, these instructions can be used to gather detailed information about what was happening on the machine at the time of the performance problem.
+
+#Required Tools#
+- **Perfcollect**: Bash script that automates data collection.
+	- Available at <http://aks.ms/perfcollect>.
+- **PerfView**: Windows-based performance tool that can also analyze trace files collected with Perfcollect.
+	- Available at <http://aks.ms/perfview>.
+
+#Preparing Your Machine#
+Follow these steps to prepare your machine to collect a performance trace.
+
+1. Download Perfcollect.
+
+	> ```bash
+	> curl -OL http://aks.ms/perfcollect
+	> ```
+
+2. Make the script executable.
+
+	> ```bash
+	> chmod +x perfcollect
+	> ```
+
+3. Install tracing prerequisites - these are the actual tracing libraries.  For details on prerequisites, see [below](#prerequisites).
+
+	> ```bash
+	> sudo ./perfcollect install
+	> ```
+
+#Collecting a Trace#
+1. Have two shell windows available - one for controlling tracing, referred to as **[Trace]** and one for running the application, referred to as **[App]**.
+2. **[App]** Setup the application shell - this enables tracing configuration inside of CoreCLR.
```

Comment:
The goal is to make these unnecessary.  There's already a plan to remove COMPlus_EnableEventLog.  I need to look into options for COMPlus_PerfMapEnabled.

I'm also looking at ways to automate this, but it requires that the script run in the same process, not a child process, so it will likely have to be sourced by the caller.  In any case, I don't have this working just yet.